clang/test/SemaCXX/cxx1z-class-template-argument-deduction.cpp

```diff
@@ -645,4 +645,27 @@ namespace undefined_warnings {
     auto test2 = TemplDObj(.0f);
   }
 }
+
+namespace GH51710 {
+template<typename T>
+struct A {
+  A(T f()) {}
+  A(int f(), T) {}
+
+  A(T array[10]) {}
+  A(int array[10], T) {}
```

Comment:
Can you add tests for incomplete array (in another struct ie

```cpp
template<typename T>
struct B {
   B(T array[]) {}
   B(int array[], T) {}
};

Reply:
Done.